src/lambda/wizards/samDeployWizard.ts

```diff
@@ -120,12 +122,15 @@ function getSingleResponse(responses: vscode.QuickPickItem[] | undefined): strin
 }
 
 export class DefaultSamDeployWizardContext implements SamDeployWizardContext {
-    public readonly onDetectLocalTemplates = detectLocalTemplates
+    public readonly templateUris: vscode.Uri[]
     public readonly getParameters = getParameters
     public readonly getOverriddenParameters = getOverriddenParameters
     private readonly helpButton = createHelpButton(localize('AWS.command.help', 'View Toolkit Documentation'))
 
-    public constructor(private readonly regionProvider: RegionProvider, private readonly awsContext: AwsContext) {}
+    public constructor(private readonly regionProvider: RegionProvider, private readonly awsContext: AwsContext) {
+        const cfnRegistry = CloudFormationTemplateRegistry.getRegistry()
+        this.templateUris = cfnRegistry.registeredTemplates.map(o => vscode.Uri.file(o.path))
```

Comment:
DefaultSamDeployWizardContext now uses the template registry instead of the old logic in  `src/test/lambda/local/detectLocalTemplates.test.ts`. 

Confirmed that this works manually, also, by invoking `AWS: Deploy SAM app` command.

Reply:
Just to confirm, what happens if you deploy a SAM app, create an additional local SAM app, and deploy that? I want to make sure that it creates a new context dynamically, otherwise the second template file won't show up.